crates/ra_assists/src/assists/add_explicit_type.rs

```diff
@@ -34,6 +35,14 @@ pub(crate) fn add_explicit_type(ctx: AssistCtx<impl HirDatabase>) -> Option<Assi
     // The binding must have a name
     let name = pat.name()?;
     let name_range = name.syntax().text_range();
+    // Assist should only be applicable if cursor is between 'let' and '='
+    let stmt_range = stmt.syntax().text_range();
+    let eq_range = stmt.syntax().descendants_with_tokens().find(|t| t.kind() == EQ)?.text_range();
```

Comment:
Let’s add an eq_token method to ast::LetStmt, so that it can be used elsewhere. There’s extensions.rs module for that

Reply:
Other than this, looks good to me!